test/parallel_api/numeric/numeric.ops/reduce_by_segment.pass.cpp

```diff
@@ -150,24 +150,24 @@ DEFINE_TEST_2(test_reduce_by_segment, BinaryPredicate, BinaryOperation)
         typedef typename ::std::iterator_traits<Iterator1>::value_type KeyT;
         typedef typename ::std::iterator_traits<Iterator2>::value_type ValT;
 
+        initialize_data(host_keys.get(), host_vals.get(), host_res_keys.get(), host_res.get(), n);
+        update_data(host_keys, host_vals, host_res_keys, host_res);
+
         std::pair<Iterator3, Iterator4> res;
         if constexpr (std::is_same_v<std::equal_to<KeyT>, std::decay_t<BinaryPredicate>> &&
                       std::is_same_v<std::plus<ValT>, std::decay_t<BinaryOperation>>)
         {
-            auto new_policy = make_new_policy<new_kernel_name<Policy, 0>>(exec);
-            res = oneapi::dpl::reduce_by_segment(std::move(new_policy), keys_first, keys_last, vals_first,
+            res = oneapi::dpl::reduce_by_segment(std::forward<Policy>(exec), keys_first, keys_last, vals_first,
```

Comment:
I made issue here: `std::forward<Policy>(exec)`
Because later you have `exec.queue().wait_and_throw();` call

Reply:
Thanks I have fixed this for this case. For the host case, we do not use exec later on so it can be forwarded.